### ISSUE TITLE
Embedding Remote File URL

### ISSUE THREAD

**Issue by KINGSABRI** [gettalong/hexapdf#66]
Hi 
Thank you for making hexapdf available 

I'm trying to create a pdf with embedding a remote file in `/F` GoToR field. 
The desire pdf structure related to this part is to be like following

```
..omitted..

   /AA <<
	   /O <<
	      /F (http://example.com/file.pdf)
		  /D [ 0 /Fit]
		  /S /GoToE
		  >>

..omitted..
```

I've tried to follow ([test_rui](https://github.com/gettalong/hexapdf/blob/3fd6283dd4a93845d4ddc0cbaea25e83fd223431/test/hexapdf/type/actions/test_uri.rb) and [test_launch](https://github.com/gettalong/hexapdf/blob/3fd6283dd4a93845d4ddc0cbaea25e83fd223431/test/hexapdf/type/actions/test_launch.rb))
```
doc = HexaPDF::Document.new
action = HexaPDF::Type::Actions::Launch.new({}, document: doc)
action.value = {F: "http://example.com/file.pdf"}
doc.write("t1.pdf")
```
It didn't work and it shouldn't, since I didn't figureout how to add the action object to doc object before generating the file

I'm sure there is something I didn't understand in the API

**Comment by gettalong** [comment]
If you want to use an action inside a PDF document, you have to make sure the document knows about it and you have to place it at the right location inside the PDF to work correctly.

So, first you have to decide how you want the action to be triggered. For example, if you want to trigger it when the fourth page gets opened, you could do the following (always jumps to fifth page when opening the fourth page):

~~~ ruby
require 'hexapdf'

doc = HexaPDF::Document.new
5.times { doc.pages.add }
doc.pages[3][:AA] = {
  O: {
    Type: :Action,
    S: :GoTo,
    D: [doc.pages[4], :Fit],
  }
}
doc.write('sample.pdf')
~~~

First we add five pages. Then we define the AdditionalActions `:AA` dictionary on the fourth page and add an action to be executed when the page gets opened (`:O`).

You can do something similar and use the required action.

Please note that not all PDF viewers honor such actions. For example, Evince and Okular on Linux completely ignore these actions.

**Comment by KINGSABRI** [comment]
Hi @gettalong 
Thank you for your explanation and swift response. 
Could you please tell me how to find these details in the documentation?

**Comment by gettalong** [comment]
You will need to consult the (freely available) [PDF 1.7 specification](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf) for most of this. There are only a handful of actions and annotations currently implemented and there are currently no convenience methods for working with them. The actions/annotations have mostly been implemented so that the [link layer](https://github.com/gettalong/hexapdf/blob/master/lib/hexapdf/layout/style.rb#L473) for boxes work.

So this means the above sample code is currently the only way to go.

**Comment by KINGSABRI** [comment]
Thanks again for your informative and swift response. 
I will close the issue as I've been answered
